docs/architecture/adr-083-out-of-band-sync.md

```diff
@@ -0,0 +1,73 @@
+# ADR 083: Supporting out of band state sync
+
+## Changelog
+
+- 2022-10-31: Initial Draft (@cmwaters)
+
+## Status
+
+Accepted
+
+## Context
+
+Whenever Tendermint begins, it initiates a handshake with the application via `Info` to gauge version compatibility and recieve the last height and app hash of the application. Tendermint then begins a replay protocol which aims to sync the heights of Tendermint's state and block store and the application's current height. 
+
+When initially designed, one invariant was that the applications current height should never exceed
+Tendermint's height. The protocol would error and the node would shut down if this were to happen. It seemed
+appropriate at the time that Tendermint which would provide blocks either through block sync or consensus
+would always be ahead. However, in light of state sync and other protocols for bootstrapping an application to 
+a particular height, this invariant is no longer necessary.
+
+## Decision
+
+Allow applications to start with a bootstrapped state alongside an empty Tendermint instance using
+a subsystem of the state sync protocol. 
+
+## Detailed Design
+
+Tendermint will perform a protocol to bootstrap to the height of the application under the following circumstances:
+
+- The returned application height is greater than Tendermint's blockstore height
+- Tendermint's block store has a height of 0 (i.e. it's empty and we're not overriding an existing instance. This check may be relaxed in the future) 
+- State sync is enabled and the config is valid (i.e it contains at least 2 rpc endpoints and a trusted header and hash)
```

Comment:
The snapshot to be used/loaded is part of this configuration?

Reply:
We assume that the snapshot has already been loaded to the application hence why the application gives us a non-zero height